orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/WaitUntilStage.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import com.netflix.spinnaker.orca.pipeline.tasks.WaitTask;
+import com.netflix.spinnaker.orca.pipeline.tasks.WaitUntilTask;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.Duration;
+import java.time.Instant;
+
+@Component
+public class WaitUntilStage implements StageDefinitionBuilder {
+
+  public static String STAGE_TYPE = "waitUntil";
+
+  @Override
+  public void taskGraph(Stage stage, TaskNode.Builder builder) {
+    builder.withTask("waitUntil", WaitUntilTask.class);
+  }
+
+  public static final class WaitStageContext {
+    private final Long waitUntil;
+    private final boolean skipRemainingWait;
+    private final Instant startTime;
+
+    @JsonCreator
+    public WaitStageContext(
+      @JsonProperty("waitUntil") @Nullable Long waitUntil,
```

Comment:
Would be nice if this had units in the name

Reply:
Will fix.